src/function/vector_string_functions.cpp

```diff
@@ -23,74 +23,69 @@ using namespace kuzu::common;
 namespace kuzu {
 namespace function {
 
-void BaseLowerUpperFunction::operation(common::ku_string_t& input, common::ku_string_t& result,
-    common::ValueVector& resultValueVector, bool isUpper) {
+void BaseLowerUpperFunction::operation(
+    ku_string_t& input, ku_string_t& result, ValueVector& resultValueVector, bool isUpper) {
     uint32_t resultLen = getResultLen((char*)input.getData(), input.len, isUpper);
     result.len = resultLen;
-    if (resultLen <= common::ku_string_t::SHORT_STR_LENGTH) {
+    if (resultLen <= ku_string_t::SHORT_STR_LENGTH) {
```

Comment:
I would suggest: refactor the function as follow:
Define a string which has the same length as input. 
Convert input to the string.
Call the stringVector::addString(resultVector, result, string).
So you don't have to add if/else for short_str, and do memcpy for overflowptr

Reply:
Do you think it would affect the performance? Because we have to write twice, from the input to the temporary string, and from the string to result.